### PR TITLE
Implemented set

### PR DESCRIPTION
**Pull request by boukeversteegh**

### CODE REVIEW

**Comment by boukeversteegh** [comment]
I removed trailing whitespace which travis complained about.

**Comment by boukeversteegh** [comment]
(bump)

**Comment by alexeykuzmin** [comment]
Hi, @boukeversteegh. Sorry for a delay, I'll review this today.

File: src/jsonpointer.js [hunk 1]
```diff
@@ -289,6 +373,10 @@
     return new Error(ERROR_MESSAGE_PREFIX + message);
   }
 
+  function setError(message) {
```
Comment (alexeykuzmin) [inline src/jsonpointer.js]: You don't need to add this method, simply use `getError` instead.

File: src/jsonpointer.js [hunk 2]
```diff
@@ -155,6 +190,55 @@
     };
   }
 
+  function createPointerSetter(target) {
+    return function(pointer, opt_targetvalue) {
+
+      if( pointer === '' ) {
```
Comment (alexeykuzmin) [inline src/jsonpointer.js]: Please keep the code style consistent.
There is always a whitespace between `if` and opening brace,
and there are no spaces before and after expression between braces.

So this line should look like this: `if (pointer === '') {`.

File: src/jsonpointer.js [hunk 3]
```diff
@@ -155,6 +190,55 @@
     };
   }
 
+  function createPointerSetter(target) {
+    return function(pointer, opt_targetvalue) {
```
Comment (alexeykuzmin) [inline src/jsonpointer.js]: Please use camelcase for names, e.g. `opt_targetValue`, not `opt_targetvalue`.

File: src/jsonpointer.js [hunk 4]
```diff
@@ -155,6 +190,55 @@
...
+      var targetkey = tokensList.splice(0, 1)[0];
+
+      // Evaluation until last object
+      while (!isUndefined(value) && !isUndefined(token = tokensList.pop())) {
+        // Let's evaluate token in current context.
+        // `getValue()` might throw an exception, but we won't handle it.
+        value = getValue(value, token);
+      }
+
+      // Retrieve last key to enforce full validation of pointer
```
Comment (alexeykuzmin) [inline src/jsonpointer.js]: Every sentence should be ended with dot.

File: README.md [hunk 5]
```diff
@@ -24,7 +24,7 @@ or copy `src/jsonpointer.js` file from repo.
 ## Include
 ###Node.js
 ```js
-var jsonpoiner = require('jsonpointer.js');  // XXX: '.js' is part of package name!
+var jsonpointer = require('jsonpointer.js');  // XXX: '.js' is part of package name!
```
Comment (alexeykuzmin) [inline README.md]: Thanks!

**Comment by alexeykuzmin** [comment]
@boukeversteegh, could use please add some tests for a new method?

**Comment by alexeykuzmin** [comment]
And README file needs to be updated with new method description.

**Pull request closed without merging**